accepted/00074-retracted-updates-support.md

```diff
@@ -0,0 +1,307 @@
+- Feature Name: Retracted updates support
+- Start Date: 2020-11-03
+- RFC PR: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+The topic of this RFC is implementing the "retracted patches" support in Uyuni.
+
+A retracted patch is a patch with attribut `status=retracted`. The purpose of
+the flag is to signalize the fact, that the patch had been released, but was
+then back (retracted) by its publisher. This can be used, for instance, when the
+publisher issues a patch that is invalid (e.g. can break a system).
+
+# Motivation
+[motivation]: #motivation
+
+- Why are we doing this?
+
+SUSE Customer Center (SCC) now exposes patches with the `retracted`
+status. Making Uyuni aware of this feature improves the SCC integration.
+
+- What use cases does it support?
+
+The main goal of the feature is to minimize the possibility of breaking user
+systems due to broken released patches.
+
+- What is the expected outcome?
+
+To make Uyuni aware of the `retracted` status and to update corresponding pieces
+of UI/XMLRPC.
+
+# Detailed design
+[design]: #detailed-design
+
+We will implement the feature in an iterative way, starting from basic support
+of the `retracted` status in the `spacewalk-repo-sync` and backend. After that,
+it should be exposed in the user-facing parts of Uyuni. Finally, we will enhance
+the Content Lifecycle feature by adding new filters operating with the
+`retracted` status.
+
+## Zypper implementation of retracted patches
+
+Understanding how zypper handles retracted patches is vital for further design
+of the feature integration.
+
+In this section, we have a system which:
+- has an installed package `hello-2.10-lp151.2.6.x86_64`
+- is registered in a repository containing a retracted patch associated with
+  package `hello-2.10-lp151.4.1.x86_64` (newer than the installed one)
+
+Operations and their outcomes:
+- listing (`zypper patches`): lists retracted patches with status `retracted`
+- searching for new versions (`zypper search -v hello`) gets us:
+  `vR | hello | package | 2.10-lp151.4.1` <--- R means retracted
+- upgrading a package (`zypper up hello`):
+  does not upgrade to a version contained in a retracted patch
+- installing specific version (`zypper in hello-2.10-lp151.4.1`):
+  **allows** installing a package contained in a retracted patch (this must be
+  taken into account, since this is the way we upgrade packages in SUMA)
```

Comment:
What's the user experience idea in this case? Should they be able to "override" the retracted status like in `zypper in`, or do we plan to be more restrictive in Uyuni?

Reply:
I've updated the draft now. I'd suggest that we display a big warning in the install/upgrade confirmation screen saying "You're about to install package(s) that belong to a retracted patch". We shouldn't prevent user from installing the packages. Maybe the problem in the retracted patch doesn't apply to them.

Reply:
Another improvement in this area would be to introduce a toggle on the packages updates list, that would hide the packages that belong to retracted patches.